src/pages/Faucet/index.tsx

```diff
@@ -38,23 +38,36 @@ export const FaucetPage = () => {
     },
     [address]
   );
+
   const onModalClose = () => {
     setIsModalOpen(false);
     setTxHash("");
   };
+
+  const readEthAmount = () => {
```

Comment:
cannot call a hook from within a function - please extract to a separate hook, also, where is the address coming from? will it always stay the same? if so, assign to a descriptive constant please, like `const FAUCET_ADDRESS` (or you could add it to the `/config` endpoint to be even cooler)